fix: set the worker timeout long enough for all workers to finish

JamesPiechota opened the pull request:
this should prevent the workers from geting a backlog of emit requests. And since each POST /tx2 has a 5 second timeout anyways the workers should move through any hung peers *relatively* quickly

JamesPiechota committed:
fix: set the worker timeout long enough for all workers to finish

this should prevent the situation where jobs that were tuck in
the message queue would get timed out, causing more jobs to be
pushed to the message queue

JamesPiechota committed:
fix: fix a bug which caused bad packing for unaligned storage modules

If a replica.2.9 storage module was configured to start at an offset that was
not aligned with the start of a partition, it would generate incorrectly
packed chunks.t

ldmberman commented on apps/arweave/src/ar_entropy_gen.erl:
```diff
@@ -228,24 +228,9 @@ do_prepare_entropy(State) ->
-    end,
+
+    %% Make sure all prior entropy writes are complete.
+    ar_entropy_storage:is_ready(StoreID),
```
What does this line do?

JamesPiechota replied:
Right now we support generating the entropy in parallel with writing it to disk (i..e while writing entropy A to disk we can start generating entropy B)

the `is_ready` call bascialy waits on the previous write to finish.

Unfortunately there have been a few race conditions due to this paralllization, so now I always wait before generting entropy which ultimately removes the parallelization.

I'd like to try to reintroduce it later, but for now it's proving problematic (e.g. when we use to just advance the cursor by ?DATA_CHUNK_SIZE the parallelization was fine, but now that we query the next synced interval we *have* to wait for th write to finish otherwise that query may be wrong and cause the same chunk to be written twice - that's the reason I removed the parallelization)

ldmberman replied:
If we take a noticeable performance hit because of that I would may be roll back "advance_offset" - it should not be needed anyway considering we store intermediate cursor values.

JamesPiechota replied:
Got it. Once I get 2.9.4 I'm planning to look more at performance - as part of that push I'll see if there's a reliable way to parallelize entropy generation and storage. RIght now I'm more worried about reliability, we've been playing a bit of whackamole the past few releases when it comes to race conditions. I think this PR gets to a pretty reliable place (at least the e2e tests pass reliably) - I'm thinking to restart from there

ldmberman commented on apps/arweave/src/ar_entropy_storage.erl:
```diff
@@ -356,6 +367,19 @@ record_entropy(ChunkEntropy, BucketEndOffset, StoreID, RewardAddr) ->
+
+%% @doc If we are not at the beginning of the entropy, shift the offset to
+%% the left. store_entropy will traverse the entire 2.9 partition shifting
+%% the offset by sector size. It may happen some sub-chunks will be written
```
> It may happen some sub-chunks will be written to the neighbouring storage module(s) on the left or on the right since the storage module may be configured to be smaller than the partition.

This is no longer relevant, right?

JamesPiechota replied:
yep! Sorry I have a edit locally that removes that comment, was just waiting on e2e tests to pass

JamesPiechota committed:
test: get the e2e tests to pass reliably

JamesPiechota committed:
fix: allow node to heal itself after not_prepared_yet

Unclear what causes this error, but when we encounter it we'll
clear out the entropy sync record so that it can be regnerated
next time
